### PR TITLE
Director counselor show/index

### PR DESCRIPTION
**Pull request by Garo-A**
Added controllers and routes for directors to see all counselors, show one counselor by ID and edit counselors.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  has_many :counselors
```
Comment (DerangedMind) [inline app/models/user.rb]: A user doesn't have many counselors. If anything it is a one-to-one, but definitely not a one-to-many!

File: app/views/director/counselors/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h1> <%= @counselor.user.first_name %> <%= @counselor.user.last_name %> </h1>
+
+<h3> Alias: </h3> <%=@counselor.alias %>
+<h3> Training Status: </h3> <%=@counselor.training %>
+<h3> Account Status: </h3> <%=@counselor.account_status %>
+<h3> Email: </h3> <%=@counselor.user.email %>
+
+<br />
+
+<%= link_to "Edit Counselor", edit_counselor_path(:id => @counselor.id) %>
```
Comment (DerangedMind) [inline app/views/director/counselors/show.html.erb]: Add a button to View All Counselors as well, please. We don't know 100% what page the user will be coming from when viewing this page, so giving them the choice to view all may be helpful to them. 😸

File: app/models/user.rb (hunk 1 above)
Comment (Garo-A) [inline app/models/user.rb]: It should; within the users table we'd have more than one user who occupies the role of "counselor", however each counselor belongs to a user.

File: app/views/director/counselors/show.html.erb (hunk 2 above)
Comment (Garo-A) [inline app/views/director/counselors/show.html.erb]: Oh, sure thing!

File: app/models/user.rb (hunk 1 above)
Comment (DerangedMind) [inline app/models/user.rb]: A single user would not have many counselors, though. We define the 1-1 relationship by adding `user_id` to our counselors table.

Comment (DerangedMind) [inline app/models/user.rb]: In addition, not all users are counselors

Comment (Garo-A) [inline app/models/user.rb]: Ahhhh shit shit yes you're right one user is only one counselor.
Apologies! Hang on.

**Pull request merged**